test/org/sagebionetworks/bridge/play/controllers/SubpopulationControllerTest.java

```diff
@@ -185,15 +190,60 @@ public void deleteSubpopulation() throws Exception {
         Http.Context context = TestUtils.mockPlayContext();
         Http.Context.current.set(context);
 
-        Result result = controller.deleteSubpopulation("AAA");
+        Result result = controller.deleteSubpopulation(SUBPOP_GUID.getGuid());
         assertEquals(200, result.status());
         String json = Helpers.contentAsString(result);
         JsonNode node = BridgeObjectMapper.get().readTree(json);
         assertEquals("Subpopulation has been deleted.", node.get("message").asText());
         
-        verify(subpopService).deleteSubpopulation(STUDY_IDENTIFIER, "AAA");
+        verify(subpopService).deleteSubpopulation(STUDY_IDENTIFIER, SUBPOP_GUID);
     }
-
+    
+    @Test(expected = UnauthorizedException.class)
+    public void getAllSubpopulationsRequiresDeveloper() throws Exception {
+        User user = new User();
+        doReturn(user).when(session).getUser(); // no developer role
+        
+        controller.getAllSubpopulations();
+        verifyNoMoreInteractions(subpopService);
```

Comment:
FYI, none of your "verifyNoMoreInteractions()" will be called, since the controller always throws before it reaches this point. (Here and in the next N tests.)